src/plugins/kibana_react/public/table_list_view/table_list_view.tsx

```diff
@@ -120,6 +131,28 @@ class TableListView<V extends {}> extends React.Component<
     this.fetchItems();
   }
 
+  componentDidUpdate(prevProps: TableListViewProps<V>, prevState: TableListViewState<V>) {
+    if (this.state.hasUpdatedAtMetadata === null && prevState.items !== this.state.items) {
```

Comment:
why is the second condition necessary?

Reply:
Because this lifecycle method is called whenever _any_ state change. Here I am only interested when the `items` state has changed (items have loaded on component did mount) and that we have some items (the array is not empty)

Reply:
Items will load in `componentDidMount`, and by then `hasUpdatedAtMetadata` will still be null. So I think just the first condition should be enough. But ok, no harm in leaving it like this.

Reply:
And if any of the component state changes _while_ the items are being fetched the method will be called. `hasUpdatedAtMetadata` will be null. And I don't want to enter the if condition at that moment. I want to wait for items to be fetched.